File: Daily/March/goMarch/LeetCode258.go
package goMarch

func AddDigits(num int) int {
	if num == 0 {
		return 0
	}
	for num >= 10 {
		sum := 0
		for num > 0 {
			sum += num % 10
			num /= 10
		}
		num = sum
	}
	return num
}

File: Daily/March/LeetCode258.cc
#include<iostream>
using namespace std;

class Solution {
public:
    int addDigits(int num) {
        if(num == 0){
            return 0;
        }
        while(num >= 10){
            int sum = 0;
            while(num > 0){
                sum += num%10; //个位数
                num /= 10; // 十位数
            }
            num = sum;
        }
        return num;
    }
};